[16.0][IMP] project_task_code: Allow search task by its code

dmitriypaulov opened the pull request:
This feature allows users to search project tasks by their codes

dreispt committed:
New module to restore State to Project Stages

sysadminmatmoz committed:
Missing templates and translations added

kellylougheed committed:
project_stage_state: Upgrade to v10

pnajman-modoolar committed:
ADD: Adding module project_key

rodrigets committed:
project_stage_state: Migration to 11.0

sbidoul committed:
remove obsolete .pot files [ci skip]

bealdav committed:
Translated using Weblate (Français) [ci skip]

Currently translated at 100,0% (8 of 8 strings)

Translation: project-11.0/project-11.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-11-0/project-11-0-project_stage_state/fr/

SalahAdDin committed:
Update tr.po

tarteo committed:
[FIX] Copying projects

mstuttgart committed:
Fix name of project.task and project.task.type model files

alexey-pelykh committed:
[MIG] project_key: Migration to 12.0

alexey-pelykh committed:
[IMP] project_key: do not regenerate project key if it's already set

anandkansagra committed:
project_stage_state: Migration to 12.0

mvrodriguez committed:
Translated using Weblate (Spanish)

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-12.0/project-12.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-12-0/project-12-0-project_stage_state/gl/

tbaden committed:
[FIX][12.0] project_key relative path

ernestotejeda committed:
project_key: add test context key

oca-travis committed:
[UPD] Update project_key.pot

marylla committed:
Added translation using Weblate (German)

OCA-git-bot committed:
[UPD] README.rst

eduardgm committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-12.0/project-12.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-12-0/project-12-0-project_stage_state/ca/

alexey-pelykh committed:
[ADD] project_task_project_required

Saran440 committed:
project_stage_state: black, isort

Saran440 committed:
project_stage_state

Saran440 committed:
[IMP] project_key: black, isort

Saran440 committed:
[13.0][MIG] project_key

OCA-git-bot committed:
pre-commit update

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-13.0/project-13.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-13-0/project-13-0-project_key/

primes2h committed:
Added translation using Weblate (Italian)

Saran440 committed:
add state in search view

oca-travis committed:
Update project_stage_state.pot

OCA-git-bot committed:
README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-13.0/project-13.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-13-0/project-13-0-project_stage_state/

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

skeller1 committed:
[FIX] project_key: avoid index out of range error on project key generation

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

OCA-git-bot committed:
project_key 13.0.1.0.1

Yoshiro009 committed:
[14.0][MIG] project_stage_state

oca-travis committed:
[UPD] Update project_stage_state.pot

OCA-git-bot committed:
project_stage_state 14.0.1.1.0

newtratip committed:
[IMP] project_key: black, isort, prettier

newtratip committed:
[MIG] project_key: Migration to 14.0

OCA-git-bot committed:
[UPD] README.rst

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/es_AR/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/fr/

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 72.7% (8 of 11 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/fr_FR/

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

Yvesldff committed:
Added translation using Weblate (French)

simahawk committed:
project_key: fix controller tests

simahawk committed:
project_key: boost tests

petrus-v committed:
try fixing CI

legalsylvain committed:
[IMP] project_task_project_required : project field mandatory by default

olgamarcocb committed:
Added translation using Weblate (Spanish)

manuelregidor committed:
[IMP] project_stage_state: black, isort, prettier

manuelregidor committed:
[MIG]project_stage_state: Migration to 15.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 50.0% (4 of 8 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/it/

yweng8111 committed:
[IMP] add function to generate unique project key automatically.

primes2h committed:
Added translation using Weblate (Italian)

eferriere committed:
[IMP] project_key: black, isort, prettier

eferriere committed:
[MIG] project_key: Migration to 15.0

eferriere committed:
The key sequences to create stories and tasks with keys, created with
a project, must be linked to the project company to avoid security
issues.

Commit:
[UPD] Update project_key.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-14.0/project-14.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_key/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (12 of 12 strings)

Translation: project-14.0/project-14.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_key/es_AR/

HekkiMelody committed:
[IMP] project_task_project_required: pre-commit execution

HekkiMelody committed:
[MIG] project_task_project_required: Migration to 14.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 83.3% (10 of 12 strings)

Translation: project-14.0/project-14.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_key/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/it/

AungKoKoLin1997 committed:
[ADD] project_task_name_with_id

Commit:
[UPD] Update project_task_project_required.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-14.0/project-14.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_task_project_required/

Commit:
[IMP] project_key: pre-commit stuff

Commit:
[MIG] project_key: Migration to 16.0

aronabencherif.diatta@xcg.africa: Update with 'model_create_multi'
decorator to allow several records to be taken at once

vincent.hatakeyama@xcg-consulting.fr: Fix _rec_names_search on project by
including 'name'. Fix task creation.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_task_project_required/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_task_project_required/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_stage_state
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_stage_state/it/

Volodiay616 committed:
[IMP] project_task_pull_request_state: add module

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_task_project_required/it/

AungKoKoLin1997 committed:
[MIG] project_task_stage_state: Migration to 16.0

Commit:
[UPD] Update project_task_stage_state.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update project_task_pull_request_state.pot

OCA-git-bot committed:
[BOT] post-merge updates

carolinafernandez-tecnativa committed:
[15.0][ADD] project_task_note: Add notes in project task
TT47764

[UPD] Update project_task_note.pot

carolinafernandez-tecnativa committed:
[IMP] project_task_note: pre-commit stuff

carolinafernandez-tecnativa committed:
[MIG] project_task_note: Migration to 16.0

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: project-16.0/project-16.0-project_task_pull_request_state
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_pull_request_state/it/

Commit:
[UPD] Update project_task_note.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: project-16.0/project-16.0-project_task_note
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_note/it/

dmitriypaulov committed:
[IMP] project_task_code: allow search task by its code

[FIX] project_task_code: fix called method name in tests

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_department
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_department/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_task_add_very_high
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_add_very_high/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (14 of 14 strings)

Translation: project-16.0/project-16.0-project_task_material
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_material/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (3 of 3 strings)

Translation: project-16.0/project-16.0-project_timeline
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timeline/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (48 of 48 strings)

Translation: project-16.0/project-16.0-project_timesheet_time_control
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timesheet_time_control/sv/

Aldeigja reviewed (approved):
Functional - LGTM!

ivs-cetmix reviewed (approved):
Code review LGTM

ivs-cetmix commented:
Hey @OCA/project-service-maintainers would be great to have this merged. Because absence of this feature affects the productivity badly 😉

alexey-pelykh commented on project_task_code/models/project_task.py:
```diff
@@ -41,3 +41,11 @@ def name_get(self):
+
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        args = args or []
```
What's the point of this line if args are just passed down?

dmitriypaulov replied:
> What's the point of this line if args are just passed down?

Args is used in super. As you see, I return combined results from super and from custom code-based domain search

alexey-pelykh replied:
I see that it's used in super, I'm just curious why you'd need to make it empty list if it's defaulted to None. I think super is capable of handling None as well, as that's the default signature of name_search, no?

dmitriypaulov replied:
> I see that it's used in super, I'm just curious why you'd need to make it empty list if it's defaulted to None. I think super is capable of handling None as well, as that's the default signature of name_search, no?

Yeah, you're right, this line is pointless, I'll remove it. Thanks for the review!

alexey-pelykh reviewed (approved):
Generally OK, one question

DemchukM commented on project_task_code/models/project_task.py:
```diff
@@ -41,3 +41,11 @@ def name_get(self):
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        args = args or []
+        domain = [("code", operator, name)]
```
You need to add a check to see if the name field is not empty.
Because of this, additional values ​​are thrown up during the search

dmitriypaulov committed:
[REF] project_task_code: remove pointless line, check if name param is set

DemchukM reviewed (approved):
LGTM

alexey-pelykh commented:
@dmitriypaulov please squash the commits into one

pilarvargas-tecnativa committed:
[IMP] project_stock_product_set: Adapt wizard to changes in product_set

Moved some of the logic shared by the wizard defined in the module
sale_product_set to the module product_set so the dependency of
project_stock_product_set should now be on product_set.

Commit:
[UPD] Update project_stock_product_set.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_stock_product_set
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stock_product_set/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: project-16.0/project-16.0-project_stock_product_set
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stock_product_set/it/

houzefa-abba committed:
[FIX] project_timeline: Fix unassigned task color

Unassigned tasks are supposed to show up in white instead of default
color, but the previous `== []` comparison is no longer working in Odoo
16 / with web_timeline 16.0.1.0.3.

This is because web_timeline forces string conversion during evaluation
of these color instructions, and an empty `user_ids` is an empty string
in this case.

This does look somewhat odd but fixes the issue short of rewriting
py-eval instructions in web_timeline, which would be a much larger task.

OCA-git-bot committed:
[BOT] post-merge updates

dreispt commented:
I think the technical implementation is incorrect.
You should inject an additional search domain tuple rather than overriding name_search...

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (11 of 11 strings)

Translation: project-14.0/project-14.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-14-0/project-14-0-project_task_project_required/es/

Commit:
[UPD] Update project_key.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_key/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_key/

vincent-hatakeyama committed:
Translated using Weblate (French)

Currently translated at 100.0% (8 of 8 strings)

Translation: project-16.0/project-16.0-project_key
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_key/fr/

victor-champonnois committed:
[ADD] project_milestone_tree

ThijsvOers committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (6 of 6 strings)

Translation: project-16.0/project-16.0-project_task_pull_request
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_pull_request/nl/

houzefa-abba committed:
[FIX] project_timeline: Fix demo task dates

This commit fixes inconsistent dates we get when building a demo DB.

This module adds a `date_start` field; this is the one we want to fill
in demo data.
We do not want to fill `date_assign`, which is an automatic field.

This is how this module used to work in v12 (was based on `date_start`).
* v12➔v13 MIG: Odoo removed the base `date_start` field so `date_start` was
  changed to `date_assign` in this module.
* v13➔v14 MIG: `date_start` was added back into this module to restore
  v12 behavior. However, demo data then got left untouched.

houzefa-abba committed:
[IMP] project_timeline: Add date constraint

Ensure `date_start` (added by this module) is always before `date_end`
(base Odoo field in the `project` module).

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update project_timeline.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_timeline
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timeline/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_timeline
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timeline/it/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_timeline
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timeline/es_AR/

Commit:
[UPD] Update project_task_name_with_id.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: project-16.0/project-16.0-project_task_name_with_id
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_name_with_id/it/

ByteMeAsap committed:
[FIX] project_role: pass company from env rather than users default company to prevent multicompany issues

OCA-git-bot committed:
[BOT] post-merge updates

augustodinizl committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_timeline
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_timeline/pt_BR/

OCA-git-bot committed:
[IMP] update dotfiles

dreispt commented on project_task_code/models/project_task.py:
```diff
@@ -41,3 +41,17 @@ def name_get(self):
         return new_result
+
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        result = []
+
+        if name:
+            domain = [("code", operator, name)]
+            tasks = self.search(domain, limit=limit)
+            result = tasks.name_get()
+
+        return (
+            super().name_search(name, args=args, operator=operator, limit=limit)
+            + result
+        )
```
Just expand the domain with the additional search condition.
Suggested pseudocode:
```suggestion
    def _name_search(self, name, args=None, operator="ilike", limit=100, name_get_uid=None):
        new_args = expression.OR([args or [], [("code", operator, name)]])
        return super()._name_search(name, new_args, operator, limit, name_get_uid)
```

Commit:
[UPD] Update project_milestone_tree.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: project-16.0/project-16.0-project_type
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_type/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: project-16.0/project-16.0-project_milestone_tree
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_milestone_tree/it/

mariancuadranetkia committed:
[FIX] project_task_parent_filter: Recover create option in "My Tasks" tree view

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_task_code
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_code/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: project-16.0/project-16.0-project_task_pull_request
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_pull_request/pt_BR/

victoralmau committed:
[IMP] project_stock: Add Forecasted Report button to stock moves from tasks

TT47932

OCA-git-bot committed:
[BOT] post-merge updates

vincent-hatakeyama committed:
[FIX] project_key: remove unnecessary French (France) translation file

Fixes #1256

Commit:
[UPD] Update project_stock.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_stock
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stock/

victoralmau committed:
[FIX] project_stock: Avoid AccessError in the form view of tasks for users without permissions in stock.move

TT49583

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (60 of 60 strings)

Translation: project-16.0/project-16.0-project_stock
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stock/pt_BR/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (60 of 60 strings)

Translation: project-16.0/project-16.0-project_stock
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stock/it/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (2 of 2 strings)

Translation: project-16.0/project-16.0-project_duplicate_subtask
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_duplicate_subtask/hr/

maisim committed:
[IMP] project_task_project_required: apply pre-commit auto fixes

maisim committed:
[MIG] project_task_project_required: Migration to 16.0

OCA-git-bot committed:
[BOT] post-merge updates

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (22 of 22 strings)

Translation: project-16.0/project-16.0-project_type
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_type/pt_BR/

pilarvargas-tecnativa committed:
Added translation using Weblate (Spanish)

pilarvargas-tecnativa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (1 of 1 strings)

Translation: project-16.0/project-16.0-project_administrator_restricted_visibility
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_administrator_restricted_visibility/es/

Commit:
[UPD] project_task_code: replaced name_search with _name_search extension

Commit:
[UPD] Update project_task_project_required.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: project-16.0/project-16.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_project_required/

grindtildeath commented on project_task_code/models/project_task.py:
```diff
@@ -41,3 +42,10 @@ def name_get(self):
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        new_args = expression.OR([args or [], [("code", operator, name)]])
+        return super()._name_search(name, new_args, operator, limit, name_get_uid)
```
```suggestion
    _rec_names_search = ["name", "code"]
```
You don't need to override `_name_search` anymore in v16.0.

cf https://github.com/odoo/odoo/pull/86588/commits/04c6859864c5ba229c1f443d7269588e348b2808

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (8 of 8 strings)

Translation: project-16.0/project-16.0-project_task_project_required
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_task_project_required/pt_BR/

eugenichafer committed:
Added translation using Weblate (Catalan)

eugenichafer committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (1 of 1 strings)

Translation: project-16.0/project-16.0-project_administrator_restricted_visibility
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_administrator_restricted_visibility/ca/

eugenichafer committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (4 of 4 strings)

Translation: project-16.0/project-16.0-project_department
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_department/ca/

eugenichafer committed:
Added translation using Weblate (Catalan)

eugenichafer committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: project-16.0/project-16.0-project_stage_extra_info
Translate-URL: https://translation.odoo-community.org/projects/project-16-0/project-16-0-project_stage_extra_info/ca/

ivs-cetmix committed:
Update project_task_code/models/project_task.py

Co-authored-by: Akim Juillerat <akim.juillerat@camptocamp.com>

ivs-cetmix committed:
[IMP] project_task_code Search by code

Search task by task code.

ivs-cetmix commented:
Hi @dreispt, thank you for your review! I have created a new branch and opened a new PR https://github.com/OCA/project/pull/1321

Could you please check it?

dreispt commented:
Superseded by #1321
